test/service/src/cli.rs

```diff
@@ -0,0 +1,343 @@
+// Copyright 2022 Parity Technologies (UK) Ltd.
+// This file is part of Cumulus.
+
+// Cumulus is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Cumulus is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Cumulus.  If not, see <http://www.gnu.org/licenses/>.
+
+use clap::{Parser, Subcommand};
+use polkadot_service::{ChainSpec, ParaId, PrometheusConfig};
+use sc_cli::{
+	CliConfiguration, DefaultConfigurationValues, ImportParams, KeystoreParams, NetworkParams,
+	Result as CliResult, RuntimeVersion, SharedParams, SubstrateCli,
+};
+use sc_service::BasePath;
+use std::{net::SocketAddr, path::PathBuf};
+
+#[derive(Debug, Parser)]
+pub struct ExportGenesisStateCommand {
+	#[clap(default_value_t = 2000u32)]
+	pub parachain_id: u32,
+
+	/// Output file name or stdout if unspecified.
+	#[clap(parse(from_os_str))]
+	pub output: Option<PathBuf>,
+
+	/// Write output in binary. Default is to write in hex.
+	#[clap(short, long)]
+	pub raw: bool,
+
+	/// The name of the chain for that the genesis state should be exported.
+	#[clap(long)]
+	pub chain: Option<String>,
+}
+
+/// Command for exporting the genesis wasm file.
+#[derive(Debug, Parser)]
+pub struct ExportGenesisWasmCommand {
+	#[clap(default_value_t = 2000u32)]
+	pub parachain_id: u32,
+
+	/// Output file name or stdout if unspecified.
+	#[clap(parse(from_os_str))]
+	pub output: Option<PathBuf>,
+
+	/// Write output in binary. Default is to write in hex.
+	#[clap(short, long)]
+	pub raw: bool,
+
+	/// The name of the chain for that the genesis wasm file should be exported.
+	#[clap(long)]
+	pub chain: Option<String>,
+}
+#[derive(Subcommand, Debug)]
+pub enum Commands {
+	#[clap(name = "export-genesis-wasm")]
+	ExportGenesisWasm(ExportGenesisWasmCommand),
+
+	#[clap(name = "export-genesis-state")]
+	ExportGenesisState(ExportGenesisStateCommand),
+
+	/// Build a chain specification.
+	BuildSpec(sc_cli::BuildSpecCmd),
+}
+
+#[derive(Debug, Parser)]
+#[clap(
+	propagate_version = true,
```

Comment:
```suggestion
	version,
	propagate_version = true,
```

Without this the application doesn't have the `version` option, thus can't use the `propagate_version`. 
Indeed it panics it the binary is built without `--release` flag

Reply:
From what I experienced, other binaries (`polkadot-collator`) doesn't have the same problem.

I think because it is using a custom build script to manually define the version (that is augmented with the commit hash:

https://github.com/paritytech/cumulus/blob/a5bfd11b5f16abde954bf204ad2de70424de0f59/polkadot-parachains/build.rs#L20

And 

https://github.com/paritytech/substrate/blob/d45e3837bc89338c6eae1ff5dc250d01e1cae0e6/utils/build-script-utils/src/version.rs#L21

Reply:
Ah I see, was not aware of that. I left it only for the test binary, should be fine now